lib/srv/discovery/access_graph.go

```diff
@@ -44,14 +42,28 @@ const (
 	batchSize = 500
 )
 
-func (s *Server) reconcileAccessGraph(ctx context.Context, currentTAGResources *aws_sync.Resources, stream accessgraphv1alpha.AccessGraphService_AWSEventsStreamClient, features aws_sync.Features) {
+var (
+	// errNoAccessGraphFetchers is returned when there are no TAG fetchers.
+	errNoAccessGraphFetchers = errors.New("no Access Graph fetchers")
+)
+
+func (s *Server) reconcileAccessGraph(ctx context.Context, currentTAGResources *aws_sync.Resources, stream accessgraphv1alpha.AccessGraphService_AWSEventsStreamClient, features aws_sync.Features) error {
 	type fetcherResult struct {
 		result *aws_sync.Resources
 		err    error
 	}
 
 	allFetchers := s.getAllAWSSyncFetchers()
-
+	if len(allFetchers) == 0 {
+		// If there are no fetchers, we don't need to continue.
+		// We will send a delete request for all resources and return.
+		upsert, toDel := aws_sync.ReconcileResults(currentTAGResources, &aws_sync.Resources{})
+		err := push(stream, upsert, toDel)
+		if err != nil {
+			s.Log.WithError(err).Error("Error pushing TAGs")
```

Comment:
This probably should be appended to the `err` instead of logged here. Otherwise, the caller cannot distinguish between those two cases.